WS2012R2/lisa/remote-scripts/ica/KVP_BasicTest.sh

```diff
@@ -1,107 +1,141 @@
+# 3. check kvp_pool file permission is 644
+# 4. check kernel version supports hv_kvp
+# 5. check lsof number for kvp after sleep 2 minutes
 ################################################################
-ICA_TESTRUNNING="TestRunning"      # The test is running
-ICA_TESTCOMPLETED="TestCompleted"  # The test completed successfully
-ICA_TESTABORTED="TestAborted"      # Error during setup of test
-ICA_TESTFAILED="TestFailed"        # Error while performing the test
+InstallLsof()
+{
+    case $DISTRO in
 
-CONSTANTS_FILE="constants.sh"
+        redhat*|centos*|fedora*)
+            yum install lsof -y
+            ;;
+        ubuntu* )
+            apt-get install -y lsof
+            ;;
+        suse* )
+            zypper install -y lsof
+                ;;
+        *)
+            msg="ERROR: Distro '$DISTRO' not supported"
+            LogMsg "${msg}"
+            UpdateSummary "${msg}"
+            SetTestStateFailed
+            exit 1
+            ;;
+    esac
 
-LogMsg()
-{
-    echo `date "+%a %b %d %T %Y"` : ${1}    # To add the timestamp to the log file
+    if [ $? -ne 0 ]; then
+        LogMsg "ERROR: Failed to install lsof"
+        UpdateSummary "ERROR: Failed to install lsof"
+        SetTestStateAborted
+        exit 1
+    fi
 }
 
-cd ~
-
-UpdateTestState()
-{
-    echo $1 > $HOME/state.txt
+dos2unix utils.sh
+# Source utils.sh
+. utils.sh || {
+    echo "Error: unable to source utils.sh!"
+    exit 1
 }
-if [ -e ~/summary.log ]; then
-    LogMsg "Cleaning up previous copies of summary.log"
-    rm -rf ~/summary.log
+#
+# Source constants file and initialize most common variables
+#
+UtilsInit
+#
+# 1. verify that the KVP Daemon is running
+#
+pid=`pgrep "hypervkvpd|hv_kvp_daemon"`
+if [ $? -ne 0 ]; then
+	LogMsg "KVP Daemon is not running by default"
+	UpdateSummary "KVP daemon not running by default, basic test: Failed"
+	SetTestStateFailed
+	exit 10
 fi
+LogMsg "KVP Daemon is started on boot and it is running"
 
-LogMsg "Updating test case state to running"
-UpdateTestState $ICA_TESTRUNNING
-
-# Source the constants file
-if [ -e ~/${CONSTANTS_FILE} ]; then
-    source ~/${CONSTANTS_FILE}
+#
+# 2. run the KVP client tool and verify that the data pools are created and accessible
+#
+uname -a | grep x86_64
+if [ $? -eq 0 ]; then
+    LogMsg "64 bit architecture was detected"
+    kvp_client="kvp_client64"
 else
-    msg="Error: no ${CONSTANTS_FILE} file"
-    echo $msg
-    echo $msg >> ~/summary.log
-    UpdateTestState $ICA_TESTABORTED
-    exit 10
+    uname -a | grep i686
+    if [ $? -eq 0 ]; then
+        LogMsg "32 bit architecture was detected"
+        kvp_client="kvp_client32"
+    else
+        LogMsg "Error: Unable to detect OS architecture"
+        SetTestStateAborted
+        exit 60
+    fi
 fi
 
+chmod +x /root/kvp_client*
+poolCount=`/root/$kvp_client | grep -i pool | wc -l`
+if [ $poolCount -ne 5 ]; then
+	LogMsg "pools are not created properly"
```

Comment:
both LogMsg and UpdateSummary messages should match, I would suggest the following:
"Error: Could not find a total of 4 KVP pools!"

Reply:
Changed based on comment, thank you.